metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/bolt/ThreatIntelJoinBolt.java

```diff
@@ -133,14 +136,18 @@ public JSONObject joinMessages(Map<String, JSONObject> streamMessageMap) {
           LOG.debug(sourceType + ": Empty rules!");
         }
 
+        // triage the threat
         ThreatTriageProcessor threatTriageProcessor = new ThreatTriageProcessor(config, functionResolver, stellarContext);
-        Double triageLevel = threatTriageProcessor.apply(ret);
+        ThreatScore score = threatTriageProcessor.apply(ret);
+
         if(LOG.isDebugEnabled()) {
           String rules = Joiner.on('\n').join(triageConfig.getRiskLevelRules());
-          LOG.debug("Marked " + sourceType + " as triage level " + triageLevel + " with rules " + rules);
+          LOG.debug("Marked " + sourceType + " as triage level " + score.getScore() + " with rules " + rules);
         }
-        if(triageLevel != null && triageLevel > 0) {
-          ret.put("threat.triage.level", triageLevel);
+
+        // attach the triage threat score to the message
+        if(score.getRuleScores().size() > 0) {
+          ret.put("threat.triage.level", toMap(score));
```

Comment:
I agree with Casey.  We need a flat structure.  I would turn this:

"threat.triage.level":{
   "score":10.0,
   "rules":[
      { 
         "name":"Abnormal DNS Port",
         "comment":null
         "score":10.0,
         "reason":"Abnormal DNS Port: expected: 53, found: 224.0.0.251:5353",
      }
   ]
}

into

"threat.triage.score": 10,
"threat.triage.rule.RuleName1.name":"Abnormal DNS Port",
"threat.triage.rule.RuleName1.comment":"Some comment...."
"threat.triage.rule.RuleName1.score": 10.0
"threat.triage.rule.RuleName1.reason": "Abnormal DNS Port: expected: 53, found: 224.0.0.251:5353"